src/jit/codegenarm64.cpp

```diff
@@ -628,18 +609,33 @@ void CodeGen::genSaveCalleeSavedRegistersHelp(regMaskTP regsToSaveMask, int lowe
     }
 
     assert((spDelta % 16) == 0);
-    assert((regsToSaveMask & RBM_FP) == 0); // We never save FP here.
 
-    // We also save LR, even though it is not in RBM_CALLEE_SAVED.
-    assert(regsToSaveCount <= genCountBits(RBM_CALLEE_SAVED | RBM_LR));
+    // We also can save FP and LR, even though they are not in RBM_CALLEE_SAVED.
+    assert(regsToSaveCount <= genCountBits(RBM_CALLEE_SAVED | RBM_FP | RBM_LR));
 
-#ifdef DEBUG
-    bool isRegsToSaveCountOdd = ((regsToSaveCount % 2) != 0);
-#endif // DEBUG
+    if (genSaveFpLrWithAllCalleeSavedRegisters)
+    {
+        // TODO: always save int regs higher than float, to be consistent?
```

Comment:
open an issue to do it after 3.0 is shipped?

Reply:
I'll open an issue, but not necessarily after 3.0 ships. I mostly didn't want to do it before this change went in, to isolate the diffs.

Reply:
https://github.com/dotnet/coreclr/issues/22457